### PR TITLE
swaynag: Truncate message to 2048 chars.

### LINKED ISSUE #2842: swaynag: Message buffer overflow.
A buffer overflow can be caused by passing a long enough message (`-m`) to swaynag. 

Example:
Repeating `swaywm` 1000 times and passing it with `-m` will cause `[buffer overflow]` to display. Run `"swaywm".repeat(1000)` in the JavaScript console to get this string.

Version:
```
swaynag version 1.0-alpha.6-264-g56c388b5 (Oct 15 2018, branch 'master')
```

### PR DESCRIPTION
**Pull request by c-edw**
This PR fixes #2842. I think this should work since `vsnprintf` truncates the `va_list` to the size specified in the second parameter. I'm probably wrong on this, so please let me know. Thanks.

### CODE REVIEW

**Comment by SirCmpwn** [comment]
If you do that I'd rather see a [...] appended to the end

**Comment by c-edw** [comment]
I have reduced the truncation limit to 128, since 2048 characters would probably never fit on the screen for anyone. It will append `...` if the message is over 128 characters.

**Comment by SirCmpwn** [comment]
That's not gonna fly. My screen can easily fit 128 characters. At time of writing my statusbar is 122 characters long and only taking up half the screen.

**Comment by c-edw** [comment]
As discussed, this will append `[buffer overflow]` instead of overwriting.

File: common/pango.c [hunk 1]
```diff
@@ -10,6 +10,9 @@
 #include "log.h"
 #include "stringop.h"
 
+#define MAX_CHARS 2048
+#define OVERFLOW "[buffer overflow]"
```
Comment (SirCmpwn) [inline common/pango.c]: Prefer static const over macros

Also, let's go with a bigger maximum, how about 16K

File: common/pango.c [hunk 2]
```diff
@@ -10,6 +10,9 @@
 #include "log.h"
 #include "stringop.h"
 
+static const char *OVERFLOW = "[buffer overflow]";
+static const int MAX_CHARS = 16384;
```
Comment (SirCmpwn) [inline common/pango.c]: lowercase

File: common/pango.c [hunk 3]
```diff
@@ -84,12 +87,12 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
...
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	static char buf[2048];
+	char buf[max_chars + sizeof(overflow)];
 
 	va_list args;
 	va_start(args, fmt);
-	if (vsnprintf(buf, 2048, fmt, args) >= 2048) {
-		strcpy(buf, "[buffer overflow]");
+	if (vsnprintf(buf, max_chars, fmt, args) >= max_chars) {
+		strcat(buf, overflow);
```
Comment (SirCmpwn) [inline common/pango.c]: Sorry to pick nits here, but can you make the size of the buffer `max_chars`, then strcpy the overflow text to the end? Something like this:

```c
char buf[max_chars];
// ...
strcpy(&buf[sizeof(buf) - sizeof(overflow)], overflow);
```

Use sizeof(buf) as well for vsnprintf.

**Comment by SirCmpwn** [comment]
Once you address the last issue, can you also squash your commits together? Thanks for your patience.

**Comment by RedSoxFan** [comment]
Thanks!

**Pull request merged**